tests/test_synced_collections/synced_collection_test.py

```diff
@@ -441,9 +521,68 @@ def set_value(sd):
 
         assert len(synced_collection) == num_threads
 
+    @pytest.mark.skipif(not NUMPY, reason="This test requires the numpy package.")
+    @pytest.mark.parametrize("dtype", NUMPY_INT_TYPES)
+    @pytest.mark.parametrize("shape", (None, (1,), (2,)))
+    def test_set_get_numpy_int_data(self, synced_collection, dtype, shape):
+        """Test setting scalar int types, which should always work."""
+        try:
+            max_value = np.iinfo(dtype).max
+        except ValueError:
+            max_value = 1
+        value = np.random.randint(max_value, dtype=dtype, size=shape)
+
+        with pytest.warns(NumpyConversionWarning):
+            synced_collection["numpy_dtype_val"] = value
+        raw_value = value.item() if shape is None else value.tolist()
+        assert synced_collection["numpy_dtype_val"] == raw_value
+
+    @pytest.mark.skipif(not NUMPY, reason="This test requires the numpy package.")
+    @pytest.mark.parametrize("dtype", NUMPY_FLOAT_TYPES)
+    @pytest.mark.parametrize("shape", (None, (1,), (2,)))
+    def test_set_get_numpy_float_data(self, synced_collection, dtype, shape):
+        """Test setting scalar float types, which work if a raw Python analog exists."""
+        # Explicitly get an array with a shape so we can
+        value = dtype(np.random.random_sample(shape))
+
+        # If casting via item does not give a base Python type, the number
+        # should fail to set correctly.
+        raw_value = value.item() if shape is None else value.tolist()
+        test_value = value[0].item() if isinstance(raw_value, list) else raw_value
+        should_fail = isinstance(test_value, (np.number, np.bool_))
```

Comment:
Rename `should_fail` to something like `has_corresponding_python_type`. It makes the reason for failure much clearer.